File: design-pattern-go/builder/text_builder.go
package builder

type TextBuilder struct {
	body string
}

func NewTextBuilder() Builder {
	return &TextBuilder{
		body: "",
	}
}

func (b *TextBuilder) MakeTitle(title string) {
	b.body += "======================================\n"
	b.body += "『"
	b.body += title
	b.body += "』\n\n"
}

func (b *TextBuilder) MakeString(str string) {
	b.body += "■"
	b.body += str
	b.body += "\n\n"
}

func (b *TextBuilder) MakeItems(items []string) {
	for _, item := range items {
		b.body += "  ・"
		b.body += item
		b.body += "\n"
	}
	b.body += "\n"
}

func (b *TextBuilder) Close() {
	b.body += "======================================\n"
}

func (b *TextBuilder) GetTextResult() string {
	return b.body
}

File: design-pattern-python/design_pattern_python/builder/text_builder.py
from typing import List

from builder import Builder, StringBuilder


class TextBuilder(Builder):
    def __init__(self) -> None:
        self.__sb = StringBuilder()

    def make_title(self, title: str) -> None:
        self.__sb.append("======================================\n")
        self.__sb.append("『")
        self.__sb.append(title)
        self.__sb.append("』\n\n")

    def make_string(self, string: str) -> None:
        self.__sb.append("■")
        self.__sb.append(string)
        self.__sb.append("\n\n")

    def make_items(self, items: List[str]) -> None:
        for item in items:
            self.__sb.append("  ・")
            self.__sb.append(item + "\n")

        self.__sb.append("\n")

    def close(self) -> None:
        self.__sb.append("======================================\n")

    def get_text_result(self) -> str:
        return self.__sb.string
